search/includes/classes/class-search.php

```diff
@@ -1030,4 +1143,23 @@ private static function query_count_incr() {
 
 		return wp_cache_incr( self::QUERY_COUNT_CACHE_KEY, 1, self::QUERY_COUNT_CACHE_GROUP );
 	}
+
+	/**
+	 * Get the maximum field count from ElasticPress and verify that it's normal
+	 *
+	 * @return {int} The maximum field count
+	 */
+	private function get_maximum_field_count() {
+		$field_count  = apply_filters( 'ep_total_field_limit', 5000 );
+
+		if ( ! is_int( $field_count ) ) {
+			$field_count = intval( $field_count );
+		}
+
+		if ( 20000 < $field_count ) {
```

Comment:
Maybe `trigger_error` or `_doing_it_wrong` to inform the user instead of just silently enforcing max value here?

Reply:
`_doing_it_wrong` is probably a good way to go :+1: